app/templates/src/js/common/text_field_selector.js

```diff
@@ -0,0 +1,26 @@
+import TomSelect from "tom-select";
+
+class TextFieldSelector {
+
+    constructor(rootElementSelector, plugins = []) {
+        this.defaultPlugins = ['checkbox_options', 'remove_button', 'clear_button']
+        this.allowedPlugins = ['dropdown_input'].concat(this.defaultPlugins);
+
+        // Filter the 'plugins' passed in to remove any plugins not part of 'allowedPlugins'
+        const validPlugins = plugins.filter(plugin => this.allowedPlugins.includes(plugin))
+
+        // Combine and deduplicate plugins from default and provided lists.
+        const allPlugins = [...new Set([...this.defaultPlugins, ...validPlugins])];
```

Comment:
Ultra-complex mechanism just to prevent a developer from using a plugin that wasn't planned. And why shouldn't he ?
Just concatenate the default plugins and the plugin submitted by the developer with the spread operator.